SingularityExecutor/src/main/java/com/hubspot/singularity/executor/SingularityExecutorMonitor.java

```diff
@@ -532,7 +533,11 @@ public void onSuccess(Integer exitCode) {
 
             message = String.format("Task killed forcibly after waiting at least %s", JavaUtils.durationFromMillis(millisWaited));
           } else if (task.wasForceDestroyed()) {
-            message = "Task killed forcibly after multiple kill requests from framework";
+            if (maybeKilledBy.isPresent()) {
+              message = String.format("Task killed forcibly by %s", maybeKilledBy.get());
```

Comment:
Because in this case the task was forcibly killed by a user, there were not necessarily multiple requests from the framework. `maybeKilledBy` is not present whe framework actions are killing a task, only when it is initiated by a user